### DISCUSSION TITLE
[zmath] Is there any way to use i32/i64 as the type for vectors/matrices?

### DISCUSSION THREAD

**Discussion by heysokam** [zig-gamedev/zig-gamedev#644] [Q&A]
I have a custom fixed point type that relies on signed integers operations for its behavior

I was wondering if I could use zmath to power its linear algebra operations, but I didn't see any `I32x4` fundamental type in the library

Are int types not supported, or did I just miss their code on my search?
If not supported, are they desirable or are they against the design of the library for some reason?

**Comment by hazeycode** [comment]
Hi. zmath was written specifically with floating point operations in mind. However, I can't think of a particular reason not to consider generalizing to integers. What do you think @michal-z ?

**Comment by a-day-old-bagel** [comment]
While we're on the topic, could f64 also be supported? Especially given the fact that zphysics is capable of running on f64, the interop between them would be nicer if zmath could be made more generic.
